src/NosCore.Controllers/Miniland/MinilandObjects/MgPacketHandler.cs

```diff
@@ -202,30 +201,30 @@ private void OpenGiftBatch()
                 if (gifts.Count > i)
                 {
                     var item = _itemProvider.Create(gifts.ElementAt(i).VNum, gifts.ElementAt(i).Amount);
-                    var inv = _clientSession.Character.InventoryService.AddItemToPocket(
+                    var inv = _clientSession!.Character.InventoryService.AddItemToPocket(
                         InventoryItemInstance.Create(item, _clientSession.Character.CharacterId));
-                    if (inv.Count != 0)
+                    if (inv != null && inv.Count != 0)
                     {
-                        _clientSession.SendPacket(_clientSession.Character.GenerateSay(
+                        await _clientSession.SendPacket(_clientSession.Character.GenerateSay(
                             $"{Language.Instance.GetMessageFromKey(LanguageKey.ITEM_ACQUIRED, _clientSession.Account.Language)}: {item.Item.Name[_clientSession.Account.Language]} x {amount}",
                             SayColorType.Green));
                     }
 
                     list.Add(new MloPmgSubPacket
-                        {BoxVNum = gifts.ElementAt(i).VNum, BoxAmount = gifts.ElementAt(i).Amount});
+                    { BoxVNum = gifts.ElementAt(i).VNum, BoxAmount = gifts.ElementAt(i).Amount });
                 }
                 else
                 {
-                    list.Add(new MloPmgSubPacket {BoxVNum = 0, BoxAmount = 0});
+                    list.Add(new MloPmgSubPacket { BoxVNum = 0, BoxAmount = 0 });
                 }
             }
 
-            ShowGifts(list.ToArray());
+            await ShowGifts(list.ToArray());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/manual/NosCoreIO/NosCore/pullRequest?prid=5262615)